[Bug]: Flaky test testNonBlockingWithMultipleMessages

### What happened?

A bug happened!
[ERROR] io.a2a.server.apps.quarkus.QuarkusA2AJSONRPCVertxTest.testNonBlockingWithMultipleMessages -- Time elapsed: 10.62 s <<< FAILURE!                                                                                                                         
  org.opentest4j.AssertionFailedError: expected: <true> but was: <false>                                                                                                                                                                                                                
          at io.a2a.server.apps.common.AbstractA2AServerTest.testNonBlockingWithMultipleMessages(AbstractA2AServerTest.java:1434)                                                                                                                                                       
          at java.base/java.lang.reflect.Method.invoke(Method.java:565)                                                                                                                                                                                                                 
          at io.quarkus.test.junit.QuarkusTestExtension.runExtensionMethod(QuarkusTestExtension.java:995)                                                                                                                                                                               
          at io.quarkus.test.junit.QuarkusTestExtension.interceptTestMethod(QuarkusTestExtension.java:843)                                                                                                                                                                              
          at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)                                                                                                                                                                                                                 
          at java.base/java.util.ArrayList.forEach(ArrayList.java:1604)        

### Relevant log output

```shell

```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

--- a/tests/server-common/src/test/java/io/a2a/server/apps/common/AbstractA2AServerTest.java
+++ b/tests/server-common/src/test/java/io/a2a/server/apps/common/AbstractA2AServerTest.java
@@ -1426,8 +1426,17 @@ public abstract class AbstractA2AServerTest {
             }
         };
 
+        // Wait for streaming subscription to be established before sending message
+        CountDownLatch streamSubscriptionLatch = new CountDownLatch(1);
+        awaitStreamingSubscription()
+                .whenComplete((unused, throwable) -> streamSubscriptionLatch.countDown());
+
         // Streaming message adds artifact-2 and completes task
         getClient().sendMessage(message2, List.of(streamConsumer), null);
+
+        // Ensure subscription is established before agent sends events
+        assertTrue(streamSubscriptionLatch.await(15, TimeUnit.SECONDS),
+                "Stream subscription should be established");
 
         // 4. Verify both consumers received artifact-2 and completion
         assertTrue(resubEventLatch.await(10, TimeUnit.SECONDS));

